ui/page/root/home_page.go

```diff
@@ -170,6 +173,133 @@ func (hp *HomePage) OnNavigatedTo() {
 	hp.isBalanceHidden = hp.AssetsManager.IsTotalBalanceVisible()
 }
 
+// initDEX initializes a new dex client if dex is not ready.
+func (hp *HomePage) initDEX() {
+	if hp.AssetsManager.DexcInitialized() {
+		return // do nothing
+	}
+
+	go func() {
+		hp.AssetsManager.InitializeDEX(hp.dexCtx)
+
+		// If all went well, the dex client must be ready.
+		dexClient := hp.AssetsManager.DexClient()
+		if dexClient == nil {
+			return // nothing to do
+		}
+
+		// Wait until dex is ready
+		<-dexClient.Ready()
+
+		hasActiveOrders := dexClient.Active()
+		if !hasActiveOrders {
+			return
+		}
+
+		// Dex has active trades, login now!
+		dexPassEditor := hp.Theme.EditorPassword(new(widget.Editor), values.String(values.StrDexPassword))
+		dexPassEditor.Editor.SingleLine, dexPassEditor.IsRequired = true, true
+
+		loginModal := modal.NewCustomModal(hp.Load).
+			Title(values.String(values.StrLoginToTradeDEX)).
+			UseCustomWidget(func(gtx C) D {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return layout.Inset{Bottom: values.MarginPadding10}.Layout(gtx, hp.Theme.Body1(values.String(values.StrLoginDEXForActiveOrders)).Layout)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return dexPassEditor.Layout(gtx)
+					}),
+				)
+			}).
+			SetPositiveButtonText(values.String(values.StrLogin)).
+			SetPositiveButtonCallback(func(isChecked bool, im *modal.InfoModal) bool {
+				dexPassEditor.SetError("")
+				err := dexClient.Login([]byte(dexPassEditor.Editor.Text()))
+				if err != nil {
+					dexPassEditor.SetError(err.Error())
+					return false
+				}
```

Comment:
We want to request wallets sync for active trades and for expired bonds **whenever** dexc login is performed. I'm proposing to modify this `Login` method to return a slice of `hotWalletIDs`. This should be walletIDs for wallets that are either involved in active trades (as we have below) or that have expired bonds that are waiting to be refunded.

Then, let's have a shared UI fn (say `ensureDEXCWalletsAreSyncing`), perhaps in the main dex page that takes this `hotWalletIDs` slice, fetches the actual wallets using `AssetsManager.WalletWithID()`, checks which ones are not yet syncing and then asks the user to start syncing for them.

In most cases, whenever we call `dexc.Login()`, we'd also need to call `ensureDEXCWalletsAreSyncing` if there are `hotWalletIDs`. The only exceptions are when we do dexc login but we don't have a `app.WindowNavigator` to use in displaying the sync prompt modal.

The reason is this:
If you login to dexc and there are expired bonds, dexc will almost immediately try to refund them. But the refund will fail if the associated wallet isn't connected to a network backend (i.e. syncing or synced). Dexc will keep retrying the refund and each trial will generate a new internal address, resulting in a number of addresses being generated but never used.

Reply:
The reason explained above should be added to the code as a comment in an appropriate part of the code, to help future code maintainers understand the rationale for the request-wallets-sync code.